libecl/src/ecl_file_view.c

```diff
@@ -0,0 +1,647 @@
+  int seqnum_index = -1;
+  if ( ecl_file_view_has_kw( ecl_file_view , INTEHEAD_KW)) {
+    const int_vector_type * intehead_index_list = hash_get( ecl_file_view->kw_index , INTEHEAD_KW );
+    int index = 0;
+    while (index < int_vector_size( intehead_index_list )) {
+      const ecl_kw_type * intehead_kw = ecl_file_view_iget_kw( ecl_file_view , int_vector_iget( intehead_index_list , index ));
+      if (ecl_rsthead_date( intehead_kw ) == sim_time) {
+        seqnum_index = index;
+        break;
+      }
+      index++;
+    }
+  }
+  return seqnum_index;
+}
+
+
+/**
+   This function will scan through the ecl_file looking for INTEHEAD
+   headers corresponding to sim_time. If sim_time is found the
+   function will return the INTEHEAD occurence number, i.e. for a
+   unified restart file like:
+
+   INTEHEAD  /  01.01.2000
+   ...
+   PRESSURE
+   SWAT
+   ...
+   INTEHEAD  /  01.03.2000
+   ...
+   PRESSURE
+   SWAT
+   ...
+   INTEHEAD  /  01.05.2000
+   ...
+   PRESSURE
+   SWAT
+   ....
+
+   The function call:
+
+   ecl_file_get_restart_index( restart_file , (time_t) "01.03.2000")
+
+   will return 1. Observe that this will in general NOT agree with the
+   DATES step number.
+
+   If the sim_time can not be found the function will return -1, that
+   includes the case when the file in question is not a restart file
+   at all, and no INTEHEAD headers can be found.
+
+   Observe that the function requires on-the-second-equality; which is
+   of course quite strict.
+
+   Each report step only has one occurence of SEQNUM, but one INTEHEAD
+   for each LGR; i.e. one should call iselect_rstblock() prior to
+   calling this function.
+*/
+
+
+bool ecl_file_view_has_sim_time( const ecl_file_view_type * ecl_file_view , time_t sim_time) {
+  int num_INTEHEAD = ecl_file_view_get_num_named_kw( ecl_file_view , INTEHEAD_KW );
+  if (num_INTEHEAD == 0)
+    return false;       /* We have no INTEHEAD headers - probably not a restart file at all. */
+  else {
+    /*
+      Should probably do something smarter than a linear search; but I dare not
+      take the chance that all INTEHEAD headers are properly set. This is from
+      Schlumberger after all.
+    */
+    int intehead_index = 0;
+    while (true) {
+      time_t itime = ecl_file_view_iget_restart_sim_date( ecl_file_view , intehead_index );
+
+      if (itime == sim_time) /* Perfect hit. */
+        return true;
+
+      if (itime > sim_time)  /* We have gone past the target_time - i.e. we do not have it. */
+        return false;
+
+      intehead_index++;
+      if (intehead_index == num_INTEHEAD)  /* We have iterated through the whole thing without finding sim_time. */
+        return false;
+    }
+  }
+}
+
+
+int ecl_file_view_seqnum_index_from_sim_time( ecl_file_view_type * parent_map , time_t sim_time) {
+  int num_seqnum = ecl_file_view_get_num_named_kw( parent_map , SEQNUM_KW );
+  int seqnum_index = 0;
+  ecl_file_view_type * seqnum_map;
+
+  while (true) {
```

Comment:
Well, I could agree, but in this example, the bottom of the while-loop is

```c
if (num_seqnum == seqnum_index)
      return -1;
```
and the `seqnum_index++` is hidden inside an `else` clause (that's always invoked except the last time).  Hence this is for all purposes a `for`-loop.

I'm not going to insist that you change anything, but I will insist that, as it is now, this is a manually driven for-loop.